block-producer/src/batch_builder/mod.rs

```diff
@@ -17,9 +17,18 @@ pub use batch::TransactionBatch;
 // BATCH BUILDER
 // ================================================================================================
 
+/// Abstraction over batch proving of transactions.
+///
+/// Transactions are aggregated into batches prior to being added to blocks. This trait abstracts
+/// over this responsibility. The trait's goal is to be implementation agnostic, allowing for
+/// multiple implementations, e.g.:
+///
+/// - in-process cpu based prover
+/// - out-of-process gpu based prover
+/// - distributed prover on another machine
 #[async_trait]
 pub trait BatchBuilder: Send + Sync + 'static {
```

Comment:
Looking at this now, I wonder if we should structure things a bit differently. Specifically, this is currently responsible for both building batches and adding them to the "batch queue" (e.g., `build_batch()` doesn't return anything). We probably should separate these into:

1. Batch builder - which takes a set of transactions and returns a transaction batch.
2. Batch queue/pool - which maintains a set of built batches.